### ISSUE TITLE
CRAM-MD5 authentication broken - encoding without a string argument

### ISSUE THREAD

**Issue by sudipm-mukherjee** [OfflineIMAP/offlineimap3#42]
#### General informations

- system/distribution (with version): Debian
- offlineimap version (`offlineimap -V`): 0.0~git20210105.00d395b+dfsg-2
- Debian Bug: https://bugs.debian.org/981385

#### Logs, error

```
2021-01-30 14:02:59 ERROR: ERROR: While attempting to sync account 'accountname'
  encoding without a string argument
2021-01-30 14:02:59 ERROR: ['  
File "/usr/share/offlineimap3/offlineimap/accounts.py", line 298, in syncrunner\n    self.__sync()\n', '  
File "/usr/share/offlineimap3/offlineimap/accounts.py", line 374, in __sync\n    remoterepos.getfolders()\n', '  
File "/usr/share/offlineimap3/offlineimap/repository/IMAP.py", line 648, in getfolders\n    imapobj = self.imapserver.acquireconnection()\n', ' 
File "/usr/share/offlineimap3/offlineimap/imapserver.py", line 592, in acquireconnection\n    self.__authn_helper(imapobj)\n', '  
File "/usr/share/offlineimap3/offlineimap/imapserver.py", line 449, in __authn_helper\n    if func(imapobj):\n', '  
File "/usr/share/offlineimap3/offlineimap/imapserver.py", line 371, in __authn_cram_md5\n    imapobj.authenticate(\'CRAM-MD5\', self.__md5handler)\n', '  
File "/usr/lib/python3/dist-packages/imaplib2.py", line 691, in authenticate\n    typ, dat = self._simple_command(\'AUTHENTICATE\', mechanism.upper())\n', '  
File "/usr/lib/python3/dist-packages/imaplib2.py", line 1684, in _simple_command\n    return self._command_complete(self._command(name, *args), kw)\n', '  
File "/usr/lib/python3/dist-packages/imaplib2.py", line 1404, in _command\n    literal = literator(data, rqb)\n', '  
File "/usr/lib/python3/dist-packages/imaplib2.py", line 2247, in process\n    ret = self.mech(self.decode(data))\n', '  
File "/usr/share/offlineimap3/offlineimap/imapserver.py", line 190, in __md5handler\n    hmac.new(bytes(passwd, encoding=\'utf-8\'), challenge,\n']
```

#### Steps to reproduce the error

- 
-

**Comment by ametzler** [comment]
Hello,
I am the reporter of the Debian bug, Sudip asked me for my .offlineimaprc file (sans private information), here it is:

[dot.offlineimaprc.txt](https://github.com/OfflineIMAP/offlineimap3/files/5899270/dot.offlineimaprc.txt)

In case it is relevant, the password is ASCII only.

TIA, Andreas

**Comment by sudipm-mukherjee** [comment]
@thekix fyi, the patch attached in https://github.com/OfflineIMAP/offlineimap3/issues/40 also fixes this issue. I have tested in my setup with a test dovecot server and also @ametzler has confirmed.
Thanks @ametzler for testing.

**Comment by sudipm-mukherjee** [comment]
Has been fixed by 96793820dec2d405b6c16efaa233f3ddf3da5e19
